Problem2_Optimal.java

```diff
@@ -0,0 +1,56 @@
+
+/**
+LeetCode Submission :
+Space Complexity : O(N^2)
+Time Complexity : O(N^2)
+Use similar logic from previous problem once we enounter a solution i.e. amount = 0 add 1 to final possible solution
```

Comment:
Time and space complexities are incorrect.

Reply:
oh  I thought it will be O(Amount * Number of Denominations available) ?

Reply:
yes that's correct.